include/seqan3/search/detail/policy_result_builder.hpp

```diff
@@ -24,52 +24,78 @@ struct policy_result_builder
+     */
+    template <typename index_cursor_t, typename configuration_t>
+    //!\cond
+        requires search_traits<configuration_t>::search_best_hits
+    //!\endcond
+    auto make_results(std::vector<index_cursor_t> internal_hits, configuration_t const &)
+    {
+        using index_t = typename index_cursor_t::index_type;
+        using position_t = std::conditional_t<index_t::text_layout_mode == text_layout::collection,
+                                              std::pair<typename index_t::size_type, typename index_t::size_type>,
+                                              typename index_t::size_type>;
+        std::vector<position_t> positions;
 
-        // output cursors or text_positions
-        if constexpr (search_traits_t::search_return_index_cursor)
+        if (!internal_hits.empty())
         {
-            return internal_hits;
+            // only one cursor is reported but it might contain more than one text position
+            auto text_pos = internal_hits[0].lazy_locate();
+            positions.push_back(text_pos[0]);
         }
-        else
-        {
-            using index_t = typename index_cursor_t::index_type;
-            using hit_t = std::conditional_t<index_t::text_layout_mode == text_layout::collection,
-                                             std::pair<typename index_t::size_type, typename index_t::size_type>,
-                                             typename index_t::size_type>;
-            std::vector<hit_t> hits;
 
-            if constexpr (search_traits_t::search_best_hits)
-            {
-                // only one cursor is reported but it might contain more than one text position
-                if (!internal_hits.empty())
-                {
-                    auto text_pos = internal_hits[0].lazy_locate();
-                    hits.push_back(text_pos[0]);
-                }
-            }
-            else
-            {
-                for (auto const & cur : internal_hits)
-                {
-                    for (auto const & text_pos : cur.locate())
-                        hits.push_back(text_pos);
-                }
-                std::sort(hits.begin(), hits.end());
-                hits.erase(std::unique(hits.begin(), hits.end()), hits.end());
-            }
+        return positions;
+    }
+
+    /*!\brief Returns a range over text positions by calling locate on each cursor.
+     * \tparam index_cursor_t The type of index cursor used in the search algorithm.
+     * \tparam configuration_t The search configuration type.
+     * \param[in] internal_hits internal_hits A range over internal cursor results.
+     * \returns a range over text positions.
+     *
+     * \details
+     *
+     * The text positions are sorted and made unique before returning them.
+     */
+    template <typename index_cursor_t, typename configuration_t>
+    auto make_results(std::vector<index_cursor_t> internal_hits, configuration_t const &)
```

Comment:
Can we add a `requires` here? It currently falls-through implicitly (if traits is not `search_return_index_cursor` and `search_best_hits`);

Is there maybe an explicit traits configuration which describes this case?

Or is this always the default? If so can you update the comment that says something like "The default behaviour is to return a range over text positions by calling locate on each cursor. If you want to limit the returned range or do not spent the time to locate on a cursor, specify ..."

Reply:
I added a requires for the output configuration (return_text_positions) and added documentation that this function is used for all search modi except search_best_hits (all, all_best, strata). I also added the requirement of return_text_positions to the `make_results` overload for search best hits. I hope this makes it clearer that the results depend on the output configuration as well as the search modus.